### ISSUE TITLE
authentication on local mongo

### ISSUE THREAD

**Issue by Blizter** [mongoeye/mongoeye#1]
hello, 

I am very honoured to post the first issue on this repository (my first issue ever on a thread anyway). Your project seems to be promising!! Also, I choose it because you gave an accessible guide to install it.

When I try to connect the mongoeye to a local mongo database, it always returns an authentication error. I verified several times the database name, collection name, user, and password. it's the right ones. Here is what the shell returned. 

``` mongoeye localhost:27017 iot sensors -u "x" -p "x"
MongoEYE DEV - MongoDB exploration tool

Connecting: Error

Failed to authenticate.
Please make sure you have entered the correct credentials.
```

Thank You for the help :-)  ,
PS : what are your plans for evolving the tool?

**Comment by michaljurecko** [comment]
Hello,

Thank you for reporting a bug.

**Please provide your MongoDB server version** using a mongo shell:
```
mongo localhost --authenticationDatabase admin -u "x" -p --eval "db.serverStatus().version"
```  

It also checks whether you can login with the same credentials as you used in mongoeye.

Thank you!

**Comment by Blizter** [comment]
Hello, 

A colleague and I followed what you said, the mongo command you gave us returned :  
```MongoDB shell version v3.4.4
connecting to: mongodb://127.0.0.1:27017/localhost
MongoDB server version: 3.4.4
```
Also, we checked our credentials several times and we managed to connect directly from the mongo shell but using the command ``` mongoeye 127.0.0.1:27017 iot sensors -u "x" -p "x" ``` returned the same error as given on the first post. Am I giving the credentials the right way?

**Comment by michaljurecko** [comment]
Hello,

I tried version 3.4.4 with Docker and it looks like everything works correctly.

In my opinion, it does not matter whether it is on localhost or other address because it successfully connects but does not log in.

Please send the exact commands you entered (mongoeye + mongo shell).

Are you sure you have a user created in the `admin` database (which is the default for mongoeye)? 
And not in another database?

The authentification database can be changed by the `--auth-db` switch.

This behavior is different from the Mongo shell that uses as auth database the database you are connecting to. It is worth considering whether it would be good to unify this behavior to prevent mistakes. I do not see any functional error yet, everything should work according to the documentation.

That's how I tested it: https://asciinema.org/a/bxivbxjxi8jtf2ftsmnv4jxio

**Comment by michaljurecko** [comment]
@Blizter  
Is that answer enough? Or does the problem persist?

**Comment by Blizter** [comment]
Sorry, I didn't had time to deal with this issue :( 
I will keep you informed when I will get back on it.

**Comment by Blizter** [comment]
Hello, 
I'm back :) Yesterday, spend my day following your indications and here is what was done step by step.
First, wrote the command ```mongoeye 127.0.0.1:27017 iot sensors -u "bi" -p "x"``` which obviously returns the error :
``` MongoEYE v0.0.1 - MongoDB exploration tool
Connecting: Error
Failed to authenticate.
Please make sure you have entered the correct credentials. 
```
Then I gave a look to mongo's logs ( ```docker-compose logs -f database```) it returned : 
```[conn758] SCRAM-SHA-1 authentication failed for bi on admin from client 172.18.0.1:49334 ; UserNotFound: Could not find user bi@admin``` 

So bi user is attached to the iot database and not admin, consequently an admin user with root role was created in the admin database and this user was called in the mongoeye command  : ```./mongoeye iot sensors -u admin -p x``` which returned the expected result : 
```
Connecting: OK
Analyzing: OK
            KEY           │ COUNT  │   %    
────────────────────────────────────────────
  all documents           │ 8      │        
  analyzed documents      │ 8      │ 100.0  
                          │        │        
  _id ➜ objectId          │ 8      │ 100.0  
  Cpu_core1_temp ➜ double │ 4      │  50.0  
  Cpu_core2_temp ➜ double │ 4      │  50.0  
  Cpu_temp ➜ double       │ 4      │  50.0  
  Internal_temp ➜ double  │ 4      │  50.0  
  src ➜ object            │ 8      │ 100.0  
  ├╴context ➜ string      │ 8      │ 100.0  
  └╴dev ➜ string          │ 8      │ 100.0  
  ts ➜ double             │ 8      │ 100.0  
  Used_memory ➜ string    │ 4      │  50.0  

OK  0.008s (local analysis)
    8/8 docs (100.0%)
    10 fields, depth 2
```
I think we are almost done, however there are few questions :  
1. does the user called in the mongeye command have to be in the admin database? If no, how do we connect without going through admin?
2. What is the minimal user role that allows mongoeye to read the fields?

**Comment by michaljurecko** [comment]
Hi,

1. You can use `--auth-db` switch to change admin database. Default auth database is `admin`. 
The user may not be from the `admin` database but must have the right to read the analyzed data. 
 **Modified example:**  `mongoeye 127.0.0.1:27017 iot sensors -u "bi" -p "x" --auth-db iot`
2. It is enough to have `read` role.

The whole problem arose probably because the behaviour of *mongoeye* is different than the behavior of *mongo* official shell. By default, official *mongo* shell uses the same database for authentication and for work. *Mongoeye*, by default, use the admin database for authentification.

Thank you for reporting this problem. I did not think about it in the development process. In the next few days I will release version v0.1, which will use the same database for work and authentication.

**Comment by Blizter** [comment]
thank you very much for your help

**Comment by michaljurecko** [comment]
Included in release [v0.1](https://github.com/mongoeye/mongoeye/releases/tag/v0.1).
